digdag-core/src/test/java/io/digdag/core/agent/OperatorManagerTest.java

```diff
@@ -88,6 +90,21 @@ public void testRunWithHeartbeatWithSuccessTask()
         verify(callback, times(0)).retryTask(any(), any(), anyInt(), any(), any());
     }
 
+    @Test
+    public void testRunWithHeartbeatWithCancelRequestedTask()
+    {
+        TaskRequest taskRequest = OperatorTestingUtils.newTaskRequest(simpleConfig).withIsCancelRequested(true);
+
+        OperatorManager om = spy(operatorManager);
+        Operator op = mock(Operator.class);
+        OperatorFactory of = mock(OperatorFactory.class);
+        doReturn(of).when(registry).get(any(), any());
+        doReturn(op).when(of).newOperator(any());
+        om.runWithHeartbeat(taskRequest);
+        verify(op, times(0)).run();
+        verify(op, times(1)).cleanup(any(TaskRequest.class));
+    }
```

Comment:
Don't need to check if `taskFailed` is called?

Reply:
I think it would be better to check, I'll add it. Thank you!

Reply:
I also added other callback cases, just in case.